egs/tedlium/s5/local/nnet3/run_tdnn.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+# this is the standard "tdnn" system, built in nnet3; it's what we use to
+# call multi-splice.
+
+# Results: 
+# %WER 15.3 | 507 17792 | 87.4 9.0 3.6 2.7 15.3 90.1 | -0.081 | exp/nnet3/tdnn_sp/decode_dev/score_10_0.5/ctm.filt.filt.sys
+# %WER 13.9 | 507 17792 | 88.4 8.0 3.6 2.3 13.9 85.8 | -0.164 | exp/nnet3/tdnn_sp/decode_dev_rescore/score_10_0.5/ctm.filt.filt.sys
+# %WER 13.8 | 1155 27512 | 88.5 8.7 2.7 2.3 13.8 84.2 | -0.076 | exp/nnet3/tdnn_sp/decode_test/score_10_0.0/ctm.filt.filt.sys
+# %WER 12.5 | 1155 27512 | 89.6 7.7 2.6 2.1 12.5 81.5 | -0.133 | exp/nnet3/tdnn_sp/decode_test_rescore/score_10_0.0/ctm.filt.filt.sys
+
+# At this script level we don't support not running on GPU, as it would be painfully slow.
+# If you want to run without GPU you'd have to call train_tdnn.sh with --gpu false,
+# --num-threads 16 and --minibatch-size 128.
+
+stage=9
+affix=
+train_stage=-10
+speed_perturb=true
+common_egs_dir=
+reporting_email=
+remove_egs=false
+decode_iter=
+
+. ./cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+suffix=
+if [ "$speed_perturb" == "true" ]; then
+  suffix=_sp
+fi
+dir=exp/nnet3/tdnn
+dir=$dir${affix:+_$affix}
+dir=${dir}$suffix
+train_set=train$suffix
+ali_dir=exp/tri3_ali$suffix
+
+local/nnet3/run_ivector_common.sh --stage $stage \
+  --speed-perturb $speed_perturb || exit 1;
+
+if [ $stage -le 9 ]; then
+  echo "$0: creating neural net configs";
+
+  # create the config files for nnet initialization
+  python steps/nnet3/tdnn/make_configs.py  \
+    --feat-dir data/${train_set}_hires \
+    --ivector-dir exp/nnet3/ivectors_${train_set} \
+    --ali-dir $ali_dir \
+    --relu-dim 500 \
+    --splice-indexes "-1,0,1 -1,0,1,2 -3,0,3 -3,0,3 -3,0,3 -6,-3,0" \
+    --use-presoftmax-prior-scale true \
+   $dir/configs || exit 1;
+fi
+
+if [ $stage -le 10 ]; then
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $dir/egs/storage ]; then
+    utils/create_split_dir.pl \
+     /export/b{09,10,11,12}/$USER/kaldi-data/egs/tedlium-$(date +'%m_%d_%H_%M')/s5/$dir/egs/storage $dir/egs/storage
+  fi
+
+  steps/nnet3/train_dnn.py --stage=$train_stage \
+    --cmd="$decode_cmd" \
+    --feat.online-ivector-dir exp/nnet3/ivectors_${train_set} \
+    --feat.cmvn-opts="--norm-means=false --norm-vars=false" \
+    --trainer.num-epochs 2 \
+    --trainer.optimization.num-jobs-initial 3 \
+    --trainer.optimization.num-jobs-final 8 \
+    --trainer.optimization.initial-effective-lrate 0.0015 \
+    --trainer.optimization.final-effective-lrate 0.00015 \
+    --egs.dir "$common_egs_dir" \
+    --cleanup.remove-egs $remove_egs \
+    --cleanup.preserve-model-interval 100 \
+    --use-gpu true \
```

Comment:
Vij, I'll take your comments in this PR into mine too :)

Xingyu

在 2016/4/6 12:03, Vijayaditya Peddinti 写道:
>
> In egs/tedlium/s5/local/nnet3/run_tdnn.sh 
> <https://github.com/kaldi-asr/kaldi/pull/644#discussion_r58648507>:
>
> > +     /export/b{09,10,11,12}/$USER/kaldi-data/egs/tedlium-$(date +'%m_%d_%H_%M')/s5/$dir/egs/storage $dir/egs/storage
> > +  fi
> > +
> > +  steps/nnet3/train_dnn.py --stage=$train_stage \
> > +    --cmd="$decode_cmd" \
> > +    --feat.online-ivector-dir exp/nnet3/ivectors_${train_set} \
> > +    --feat.cmvn-opts="--norm-means=false --norm-vars=false" \
> > +    --trainer.num-epochs 2 \
> > +    --trainer.optimization.num-jobs-initial 3 \
> > +    --trainer.optimization.num-jobs-final 8 \
> > +    --trainer.optimization.initial-effective-lrate 0.0015 \
> > +    --trainer.optimization.final-effective-lrate 0.00015 \
> > +    --egs.dir "$common_egs_dir" \
> > +    --cleanup.remove-egs $remove_egs \
> > +    --cleanup.preserve-model-interval 100 \
> > +    --use-gpu true \
>
> I think there is no point explicitly specifying use-gpu true as we 
> never train without GPUs.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/kaldi-asr/kaldi/pull/644/files/036396efc35ad2cdb5b2039fdfce6b8e9f3d5786#r58648507>
>